enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultControllerEnricher.java

```diff
@@ -97,11 +97,26 @@ public DefaultControllerEnricher(MavenEnricherContext buildContext) {
     @Override
     public void addMissingResources(KubernetesListBuilder builder) {
         final String name = getConfig(Config.name, MavenUtil.createDefaultResourceName(getContext().getGav().getSanitizedArtifactId()));
-        final ResourceConfig config = new ResourceConfig.Builder()
+        ResourceConfig resourceConfig = getConfiguration().getResource().orElse(null);
+        ResourceConfig config = null;
+        if(resourceConfig != null) {
+            config = new ResourceConfig.Builder()
                     .controllerName(name)
                     .imagePullPolicy(getConfig(Config.pullPolicy))
+                    .volumes(resourceConfig.getVolumes())
+                    .withLiveness(resourceConfig.getLiveness())
+                    .withReadiness(resourceConfig.getReadiness())
+                    .withConfigMap(resourceConfig.getConfigMap())
+                    .withServiceAccount(resourceConfig.getServiceAccount())
                     .withReplicas(Configs.asInt(getConfig(Config.replicaCount)))
                     .build();
+        } else {
+            config = new ResourceConfig.Builder()
+                    .controllerName(name)
+                    .imagePullPolicy(getConfig(Config.pullPolicy))
+                    .withReplicas(Configs.asInt(getConfig(Config.replicaCount)))
+                    .build();
+        }
```

Comment:
would do sth like which makes it less redundant

```java
configBuilder = new ResourceConfig.Builder().controllerName(name)...;
if (resourceConfig != null) {
  configBuilder.volumes(...)...;

}

config = configBuilder.build();
```

Reply:
Sorry to bother you again, but what do you think about using a `ResourceConfig.Builder(ResourceConfig)` constructor which makes a deep copy of the given argument and then uses this as the internal resource object which is handed out later. That way, we can add new fields to ResourceConfig without bothering of enrichers using it (they get automatically the new fields initialized).
https://github.com/fabric8io/docker-maven-plugin/blob/38eba5ffa6920ea0bd24a1cc03c5c7224a26d4f4/src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java#L323-L329 how this is done for the BuildConfiguration.


See e.g.